### PR TITLE
Implement convert_component! for StandardLoad <- PowerLoad

### PR DESCRIPTION
**Pull request by GabrielKS**
This will be used for a new system in PowerSystemCaseBuilder. Includes unit tests.
---

 - Define convert_component! for conversions from PowerLoad to StandardLoad
 - Add unit tests for this method along the lines of existing convert_component! unit tests
 - Refactor convert_component! unit tests to remove repetition

### CODE REVIEW

File: src/base.jl [hunk 1]
```diff
@@ -1991,6 +1991,39 @@ function convert_component!(
...
 
+"""
+Converts a PowerLoad component to a StandardLoad component and replaces the original in the
+system. Does not set any fields in StandardLoad that lack a PowerLoad equivalent
+"""
+function convert_component!(
+    new_type::Type{StandardLoad},
+    old_load::PowerLoad,
+    sys::System;
+    kwargs...,
```
Comment (jd-lara) [inline src/base.jl:2002]: In Julia the argument after the `!` is the one being modified. In this case is the system that is being changed. 

```suggestion
function convert_component!(
    sys::System,
    old_load::PowerLoad
    new_type::Type{StandardLoad},
    ;
    kwargs...,
```

**Review by jd-lara — CHANGES_REQUESTED** [review]
Some changes are required.

File: src/base.jl (hunk 1 above)
Comment (GabrielKS) [reply src/base.jl:2002]: The existing `convert_component!` methods have the new type as the first argument, then the old component, then the system: https://github.com/NREL-Sienna/PowerSystems.jl/blob/1b7c1c0c4c65a11808640de2b556157a062bd2ee/src/base.jl#L1922-L1927 and https://github.com/NREL-Sienna/PowerSystems.jl/blob/1b7c1c0c4c65a11808640de2b556157a062bd2ee/src/base.jl#L1956-L1961 and I was maintaining consistency with those. Should I change those existing methods to follow the Julia convention?

File: test/test_powersystemconstructors.jl [hunk 2]
```diff
@@ -132,26 +132,51 @@ end
 end
 
 @testset "Test component conversion" begin
-    test_conversion =
+    # Reusable resources for this testset
+    load_test_system = () -> System(joinpath(BAD_DATA, "case5_re.m"))
```
[the remark this answers is not in the document]
Comment (GabrielKS) [reply test/test_powersystemconstructors.jl]: Done, reverted the old test and now use PSB for the new one.

File: src/base.jl (hunk 1 above)
Comment (GabrielKS) [reply src/base.jl]: Yes, I should. Fixed in https://github.com/NREL-Sienna/PowerSystems.jl/pull/1022/commits/703762a15651373fff3ddf1905aad34917b30934.

File: test/test_powersystemconstructors.jl [hunk 3]
```diff
@@ -167,5 +167,33 @@ end
...
+            dates = collect(
+                Dates.DateTime("2020-01-01T00:00:00"):Dates.Hour(1):Dates.DateTime(
+                    "2020-01-01T23:00:00",
+                ),
+            )
+            data = collect(1:24)
+            ta = TimeSeries.TimeArray(dates, data, [component_name])
+            time_series = SingleTimeSeries(; name = ts_name, data = ta)
+            add_time_series!(sys, old_component, time_series)
+            @test get_time_series(SingleTimeSeries, old_component, ts_name) isa SingleTimeSeries
```
[the remark this answers is not in the document]
Comment (GabrielKS) [reply test/test_powersystemconstructors.jl]: Fixed in https://github.com/NREL-Sienna/PowerSystems.jl/pull/1022/commits/703762a15651373fff3ddf1905aad34917b30934

File: test/test_powersystemconstructors.jl [hunk 4]
```diff
@@ -167,5 +167,33 @@ end
...
+
+            convert_component!(StandardLoad, old_component, sys)
+            @test isnothing(get_component(typeof(old_component), sys, component_name))
+            new_component = get_component(StandardLoad, sys, component_name)
+            @test !isnothing(new_component)
+            @test get_name(new_component) == component_name
+            @test get_time_series(SingleTimeSeries, new_component, ts_name) isa SingleTimeSeries
```
[the remark this answers is not in the document]
Comment (GabrielKS) [reply test/test_powersystemconstructors.jl]: Fixed in https://github.com/NREL-Sienna/PowerSystems.jl/pull/1022/commits/703762a15651373fff3ddf1905aad34917b30934

File: test/test_powersystemconstructors.jl [hunk 5]
```diff
@@ -167,5 +167,33 @@ end
...
+            # Conversion back is not implemented
+        end
+
```
[the remark this answers is not in the document]
Comment (GabrielKS) [reply test/test_powersystemconstructors.jl]: Fixed in https://github.com/NREL-Sienna/PowerSystems.jl/pull/1022/commits/703762a15651373fff3ddf1905aad34917b30934

**Review by jd-lara — COMMENTED** [review]
@GabrielKS you need to run the formatter

**Review by jd-lara — APPROVED** [review]

**Pull request merged**